mapillary/utils/filter.py

```diff
@@ -371,3 +373,91 @@ def compass_angle(data: list, angles: tuple = (0.0, 360.0)) -> list:
         for feature in data
         if angles[0] <= feature["properties"]["compass_angle"] <= angles[1]
     ]
+
+
+def is_looking_at(image_feature: Feature, look_at_feature: Feature) -> bool:
+    """Return whether the image_feature is looking at the look_at_feature
+
+    :param image_feature: The feature set of the image
+    :type image_feature: dict
+
+    :param look_at_feature: The feature that is being looked at
+    :type look_at_feature: dict
+
+    :return: Whether the diff is greater than 310, or less than 50
+    :rtype: bool
+    """
+
+    # Pano accessible via the `get_image_layer`
+    # in config/api/vector_tiles.py
+    if image_feature["properties"]["is_pano"]:
+        return True
+
+    # Compass angle accessible via the `get_image_layer`
+    # in config/api/vector_tiles.py
+    if image_feature["properties"]["compass_angle"] < 0:
+        return False
+
+    # Getting the difference between the two provided GeoJSONs and the compass angle
+    diff: int = (
+        abs(
+            bearing(start=image_feature, end=look_at_feature)
+            - image_feature["properties"]["compass_angle"]
+        )
+        % 360
+    )
+
+    # If diff > 310 OR diff < 50
+    return 310 < diff or diff < 50
```

Comment:
Can you explain this snippent?

Reply:
This function is a `Python` implementation of the `TypeScript` code snippet functionality that Chris shared in #14. Extracting from that code snippet was this function,

```typescript
function isLookingAt(imageFeature: any, lookatFeature: any): boolean {
    if (imageFeature.properties.pano) {
        return true;
    }

    // TODO ca < 0 or ca === -1?
    if (imageFeature.properties.ca < 0) {
        return false;
    }

    const bearing: number = Math.abs(turf.bearing(imageFeature, lookatFeature) - imageFeature.properties.ca) % 360;
    return 310 < bearing || bearing < 50;
}
```

The above code snippet replicates the same functionality as the aforementioned code.